apps/aecore/src/aec_vm_chain.erl

```diff
@@ -229,7 +236,15 @@ oracle_query(Oracle, Q, Value, QTTL, RTTL,
         {error, _} = E -> E
     end.
 
-oracle_respond(Oracle, QueryId, Signature, Response, ResponseTTL,
+-spec oracle_respond(aec_keys:pubkey(), aeo_query:id(), binary(), binary(),
+                     aeo_oracles:relative_ttl(), chain_state()) ->
+    prim_op_result().
+oracle_respond(Oracle, QueryId, Signature, Response, ResponseTTL, State) ->
+    on_chain_only(State,
```

Comment:
Is spend allowed?

Reply:
Spend is allowed. If it is performed to an on-chain addres - on-chain changes are discarded